packages/tokens/scripts/extract-carbon-parts/convertObjectToDtcgFormat.ts

```diff
@@ -0,0 +1,209 @@
+
+import type { CarbonDesignTokens, CarbonDesignToken } from './@types/CarbonDesignTokens.d.ts'
+
+import { baseFontSize } from '@carbon/layout';
+
+export const dimensionRegex: RegExp = /^([+-]?\d+(?:\.\d+)?)(?:(px|rem|em|%|vw|vh|vmin|vmax)|)$/;
+
+export function convertObjectToDtcgFormat({ value, type, group }: Args): CarbonDesignTokens | CarbonDesignToken {
+  return recursivelyProcessObject({ value, type, group });
+}
+
+function recursivelyProcessObject({ key, value, type, group}: Args): CarbonDesignTokens | CarbonDesignToken | Record<string, any> {
+  if (typeof value === 'object' && value !== null && !Array.isArray(value)) {
+    // recursively process each key
+    const result: Record<string, any> = {};
+    for (const key in value) {
+      // for better safety (says Copilot)
+      if (Object.prototype.hasOwnProperty.call(value, key)) {
+        result[key] = recursivelyProcessObject({ key, value: value[key], type, group });
+      }
+    }
+    return result;
+  } else {
+    // this is a leaf with a string or number value => convert to DTCG format
+    if (typeof value === 'string' || typeof value === 'number') {
+      switch (type) {
+        case 'color':
+          return {
+            '$type': 'color',
+            '$value': value,
+            'group': group || 'cds-generic-color',
+            'private': true,
+            'cds-original-value': value
+          };
+        case 'cubic-bezier':
+          return {
+            // TODO convert to `cubicBezier` per DTCG specifications when we are sure that Style Dictionary process it correctly
+            // see: https://www.designtokens.org/tr/drafts/format/#cubic-bezier
+            '$type': 'cubic-bezier',
+            '$value': value,
+            'group': group || 'cds-generic-cubic-bezier',
+            'private': true,
+            'cds-original-value': value
+          };
+        case 'duration':
+          const convertedDurationValue = convertDurationValue(value);
+          if (convertedDurationValue !== undefined) {
+            return {
+              '$type': 'duration',
+              '$value': convertedDurationValue.$value,
+              'unit': convertedDurationValue.unit,
+              'group': group || 'cds-generic-duration',
+              'private': true,
+              'cds-original-value': value
+            };
+          } else {
+            const unknownDurationToken = returnUnknownToken(value, `🚨 convertDurationValue: value for key "${key}" / group "${group}" is not in the expected format:`);
+            return unknownDurationToken;
+          }
+        case 'font-family':
+          return {
+            '$type': 'font-family',
+            '$value': value,
+            'group': group || 'cds-generic-font-family',
+            'private': true,
+            'cds-original-value': value
+          };
+        case 'font-size':
+          const convertedFontSizeValue = convertSizeValue(value, false);
```

Comment:
```suggestion
          const convertedFontSizeValue = convertSizeValue(value);
```
[Issue] By passing in `false` the outputted values are in rems not px, but the unit is still px. This is causing incorrect font-size styles to be generated.

Example: `carbon-extracted/typography/styles.json`

```
"font-size": {
  "$type": "font-size",
  "$value": 0.875,
  "unit": "px",
  "group": "cds-typography",
  "private": true,
  "cds-original-value": "0.875rem"
},
```

Reply:
It was actually an issue/bug in the `convertSizeValue()` function, see https://github.com/hashicorp/design-system/pull/3169/commits/7fb9dd95e5d95b39abfe65757d2ee10cb724af73